README.md

```diff
@@ -100,11 +97,46 @@ package { 'PSExcel-psc':
 }
 ```
 
+### The provider
+
+The provider to use will be either `windowspowershell` or `powershellcore`. Nodes using `powershell.exe` will use `windowspowershell`, and nodes that have PowerShell core (`pwsh.exe`) will use the `powershellcore` provider with both the `psrepository` and `package` types.
+
 ## Limitations
 
 Note that PowerShell modules can be installed side by side so installing a newer
 version of a module will not remove any previous versions.
 
+## Reference
+
+### Types
+
+### psrepository
+
+Allows you to specify and configure a repository. The type expects a valid OneGet package provider source over an HTTP or HTTPS url.
+
+#### Properties/Parameters
+
+#### `name`
+
+The name of the gallery to register on the computer. Must be unique. Cannot use `PSGallery` as the value for this property.
+
+#### `source_location`
+
+The url to the repository that you would like to register. Must be a valid HTTP or HTTPS url. This url will be used for the underlying `SourceLocation` property and will be used as the base url for `PublishLocation`, `ScriptSourceLocation`, `ScriptPublishLocation`. Cannot use the same url as the default PSGallery.
```

Comment:
Should it be `PSGallery`? Like how you wrote it above.

Reply:
I'd leave it out of the code blocks here. It is the titular name of the public, Microsoft run repository as well as the general name given to such repositories.  We could swap this instance out with just 'gallery', potentially.

Reply:
Possibly replace for clarity to: